snippets/src/lib.rs

```diff
@@ -9,112 +9,414 @@
 //! # Feature flags
 //!
 //! The main library, which enables consumers to plug their own implementations, is available by default.
-//! Features are most commonly used to enable support for languages.
+//! Features are most commonly used to enable support for languages,
+//! but other kinds of flags exist; see the table below for details.
 //!
 //! Name | Description | Kind
 //! ---|---|---
-//! `all-languages` | Enables all features that are of the kind "Language" | Language
-//! `c` | Enables support for C | Language
-//! `cpp` | Enables support for C++ | Language
-//!
-//! # Examples
-//!
-//! (TODO)
+//! `lang-all` | Enables all features that are of the kind "Language" | Language
+//! `lang-c99-tc3` | Enables support for C99 TC3 | Language
+//! `sha2-asm` | Enables hardware acceleration for SHA2 | Performance
 
-use std::{cmp::Ordering, io::Read};
+use std::{
+    borrow::Cow,
+    cmp::Ordering,
+    marker::PhantomData,
+    ops::{Range, RangeInclusive},
+    str::Utf8Error,
+};
 
+use base64::Engine;
 use derivative::Derivative;
-use derive_more::{Deref, Index};
-use fallible_iterator::FallibleIterator;
+use derive_more::{Constructor, Deref, From, Index};
+pub use fallible_iterator::FallibleIterator;
 use flagset::{flags, FlagSet};
 use getset::{CopyGetters, Getters};
+use itertools::Itertools;
+use once_cell::sync::OnceCell;
+use sha2::{Digest, Sha256};
 use strum::{Display, EnumIter};
+use tap::Pipe;
 use thiserror::Error;
 use typed_builder::TypedBuilder;
 
+pub mod debugging;
+pub mod language;
+
+/// Convenience import for all types that
+/// an implementation of [`Extractor`] would likely need.
+///
+/// Some commonly-named types are renamed to reduce the liklihood of collisions
```

Comment:
[Spelling]:

```suggestion
/// Some commonly-named types are renamed to reduce the likelihood of collisions
```

"likelihood" is such a weird looking word!